src/c++/simulator/geometry.cpp

```diff
@@ -0,0 +1,242 @@
+	y += other.y; 
+	z += other.z;
+	return *this;
+}
+
+Point Point::operator-(const Point& other) const
+{
+	return Point(x - other.x, y - other.y, z - other.z);
+}
+
+Point Point::operator*(const double& other)
+{
+	return Point(x * other, y * other, z * other);
+}
+
+Point Point::operator/(const double& other)
+{
+	return Point(x / other, y / other, z / other);
+}
+
+bool Point::operator==(const Point& other)
+{
+	return (x == other.x && y == other.y && z == other.z);
+}
+
+Point& Point::operator=(const Point& other)
+{
+	if (this == &other) return *this;
+	std::memcpy(this, &other, sizeof(Point));
+	return *this;
+}
+
+double Point::norm()
+{
+	return std::sqrt((x)*(x) + (y)*(y) + (z)*(z));
+}
+
+double Point::dot(const Point& other) const
+{
+	return x*other.x + y*other.y + z*other.z;
+}
+
+Point Point::cross(const Point& other) const
+{
+	return Point(y*other.z - z*other.y, z*other.x - x*other.z, x*other.y - y*other.x );
+}
+
+void Point::print() const 
+{
+	std::cout << "{ " << x << ", " << y << ", " << z << " }" <<std::endl;;
+}
+
+double Geometry::get_volume()
+{
+	return 0.0;
+}
+
+double Geometry::get_surface()
+{
+	return 0.0;
+}
+
+double Geometry::get_radius()
+{
+	return 0.0;
+}
+
+double PrimitiveGeometry::get_volume()
+{
+	return 0.0;
+}
+
+double PrimitiveGeometry::get_surface()
+{
+	return 0.0;
+}
+
+double PrimitiveGeometry::get_radius()
+{
+	return 0.0;
+}
+
+InfinityPlane::InfinityPlane(double x, double y, double z, double nx, double ny, double nz)
+	: point(Point(x, y, z))
+	, normal(Vector(nx, ny, nz))
+{
+}
+
+Vector InfinityPlane::get_unitary_normal()
+{
+	return normal * (1.0 / normal.norm());
+}
+
+double InfinityPlane::get_volume()
+{
+	return 0.0;
+}
+
+double InfinityPlane::get_surface()
+{
+	return 0.0;
+}
+
+Sphere::Sphere(double x, double y, double z, double radius)
+	: center(Point{x, y, z})
+	, radius(radius)
+{
+}
+
+Sphere::Sphere(double radius)
+	: center(Point{0.0, 0.0, 0.0})
+	, radius(radius)
+{
+}
+
+double Sphere::get_volume()
+{
+	return get_surface() * radius / 3.0;
+}
+
+double Sphere::get_surface()
+{
+	return 4.0 * M_PI * radius * radius;
+}
+
+double Sphere::get_radius()
+{
+	return radius;
+}
+
+Edge::Edge(Point p1, Point p2)
+{
+	points.push_back(p1);
+	points.push_back(p2);
+}
+
+Vector Edge::get_vector()
+{
+	return points.at(1) - points.at(0);
+}
+
+Vector Edge::get_unitary_direction()
+{
+	Vector v = get_vector();
+	return v/v.norm();
+}
+
+double Edge::get_volume()
+{
+	return 0.0;
+}
+
+double Edge::get_surface()
+{
+	return 0.0;
+}
+
+Vector PlanarGeometry::get_normal()
+{
+	return edges.at(0).get_vector().cross(edges.at(1).get_vector());
+}
+
+Vector PlanarGeometry::get_unitary_normal()
+{
+	Vector v = get_normal();
+	return v/v.norm();
+}
+
+double PlanarGeometry::get_volume()
+{
+	return 0.0;
+}
+
+Square::Square(Point p1, Point p2, Point p3, Point p4)
+{
+	points.push_back(p1);
+	points.push_back(p2);
+	points.push_back(p3);
+	points.push_back(p4);
+	
+	edges.push_back(Edge(p2, p1));
```

Comment:
I believe we could remove those edges. We can track them following some convention, for example, using the right-hand rule. In this way, we don't have to allocate another `std::vector`.
What do you think?